### ISSUE TITLE
People Picker not working for Host Name Site collections

### ISSUE THREAD

**Issue by dsohan** [Yvand/EntraCP#300]
Entra CP works for Central admin and Root site collection.  However under that same web app we have a host name site collection, however the People Picker just throws "Server not accessible" on the people picker.

Looking at the ULS logs, the error we see is the following: 

Leaving Monitored Scope: ([EntraCP] Querying Microsoft Entra ID tenant 'contoso.onmicrosoft.com' for users and groups, with input 'carlos') Execution Time=14375.1887; CPU Milliseconds=1; SQL Query Count=0; Parent=[EntraCP] Send request to "contoso.onmicrosoft.com" based on input "carlos"	 
12/06/2024 13:59:06.74 	w3wp.exe (0x0FC8)                       	0x2548	EntraCP                       	Lookup                        	1337	Unexpected	[EntraCP] Unexpected error : **Could not authenticate for tenant 'contoso.onmicrosoft.com': AuthenticationFailedException: ClientCertificateCredential authentication failed: Retry failed after 3 tries. Retry settings can be adjusted in ClientOptions.Retry or by configuring a custom retry policy in ClientOptions.RetryPolicy.**	 
12/06/2024 13:59:06.74 	w3wp.exe (0x0FC8)                       	0x2548	EntraCP                       	Lookup                        	1337	Medium  	[EntraCP] Got 0 users/groups in 14375 ms from 'contoso.onmicrosoft.com' with input 'carlos'	 
12/06/2024 13:59:06.74 	w3wp.exe (0x0FC8)                       	0x2548	SharePoint Foundation         	Monitoring                    	b4ly	High    	Leaving Monitored Scope: ([EntraCP] Send request to "contoso.onmicrosoft.com" based on input "carlos") Execution Time=14375.3066; CPU Milliseconds=1; SQL Query Count=0; Parent=[EntraCP] Total time spent to query Microsoft Entra ID tenant(s)	 
12/06/2024 13:59:06.74 	w3wp.exe (0x0FC8)                       	0x2548	SharePoint Foundation         	Monitoring                    	b4ly	High    	Leaving Monitored Scope: ([EntraCP] Total time spent to query Microsoft Entra ID tenant(s)) Execution Time=14375.4383; CPU Milliseconds=1; SQL Query Count=0; Parent=SPClaimProvider.FillSearch()

**Comment by Yvand** [comment]
hi @dsohan, this is strange. I see it failed after 14 secs,  could it be a firewall blocking this request?
Can you possibly try to copy the standalone page TEMPLATE/ADMIN/EntraCP/TroubleshootEntraCP.aspx somewhere in your SharePoint LAYOUTS folder, to be able to call it from the web app, and test if it is able to retrieve the access token?
Please note that you need to edit this page using a text editor to set the variables with values that fit your environment.

**Comment by dsohan** [comment]
I would say yes to a firewall if it wasn't working on the root site collection or in Central admin,

but it's weird that it seems to only be a problem on Host name site collections.  Regular site collections seem to be okay.

Denesh

________________________________

**Comment by Yvand** [comment]
Indeed, if it is working on the root site collection, a firewall seems unlikely.
Can you test using the page TroubleshootEntraCP.aspx as I mentioned in my previous message?

**Comment by dsohan** [comment]
Hi Yvand,
As an update, it seems that after reconfiguring the authentication provider in SharePoint, EntraCP started working. We're still having an error on one of the server in the farm, but we're having other issues on that server too.  So we're going to replace it in the next few weeks in hopes it resolves the issue.

Thank you for your replies as we can close this issue.

Regards,
Denesh

________________________________
